web-common/src/features/models/workspace/ModelEditor.svelte

```diff
@@ -141,230 +85,29 @@
-      });
-
-      for (let e of tr.effects)
-        if (e.is(addUnderline)) {
-          underlines = underlines.update({
-            add: [underlineMark.range(e.value.from, e.value.to)],
-          });
-        }
-      return underlines;
-    },
-    provide: (f) => EditorView.decorations.from(f),
-  });
-
-  onMount(() => {
-    editor = new EditorView({
-      state: EditorState.create({
-        doc: blob,
-        extensions: [
-          editorTheme(),
-          lineNumbers(),
-          highlightActiveLineGutter(),
-          highlightSpecialChars(),
-          history(),
-          drawSelection(),
-          dropCursor(),
-          EditorState.allowMultipleSelections.of(true),
-          indentOnInput(),
-          syntaxHighlighting(defaultHighlightStyle, { fallback: true }),
-          bracketMatching(),
-          closeBrackets(),
-          autocompleteCompartment.of(
-            makeAutocompleteConfig(schema, defaultTable),
-          ), // a compartment makes the config dynamic
-          rectangularSelection(),
-          highlightActiveLine(),
-          highlightSelectionMatches(),
-          keymap.of([
-            ...closeBracketsKeymap,
-            ...defaultKeymap,
-            ...searchKeymap,
-            ...historyKeymap,
-            ...completionKeymap,
-            ...lintKeymap,
-            indentWithTab,
-          ]),
-          Prec.high(
-            keymap.of([
-              {
-                key: "Enter",
-                run: insertNewline,
-              },
-            ]),
-          ),
-          Prec.highest(
-            keymap.of([
-              {
-                key: "Tab",
-                run: acceptCompletion,
-              },
-            ]),
-          ),
-          sql({ dialect: DuckDBSQL }),
-          keymap.of([indentWithTab]),
-          EditorView.updateListener.of((v) => {
-            if (v.focusChanged && v.view.hasFocus) {
-              dispatch("receive-focus");
-            }
-            if (v.docChanged) {
-              localContent = v.state.doc.toString();
-
-              if (autoSave) saveContent();
-            }
-          }),
-        ],
-      }),
-      parent: editorContainerComponent,
-    });
-  });
-
-  // REACTIVE FUNCTIONS
-
-  function updateEditorContents(newContent: string) {
-    if (editor && !editor.hasFocus) {
-      let curContent = editor.state.doc.toString();
-      if (newContent != curContent) {
-        // TODO: should we debounce this?
-        editor.dispatch({
-          changes: {
-            from: 0,
-            to: curContent.length,
-            insert: newContent,
-          },
-        });
-      }
-    }
-  }
-
   function updateAutocompleteSources(
     schema: { [table: string]: string[] },
     defaultTable?: string,
   ) {
     if (editor) {
-      editor.dispatch({
-        effects: autocompleteCompartment.reconfigure(
-          makeAutocompleteConfig(schema, defaultTable),
-        ),
+      queueMicrotask(() => {
```

Comment:
**Issue**: Using 'queueMicrotask' to dispatch editor updates can lead to uncontrolled execution timing, potentially causing race conditions or UI inconsistencies, especially in a complex UI environment where multiple tasks might be queued simultaneously. <br> **Fix**: Replace 'queueMicrotask' with a more controlled asynchronous handling mechanism such as 'setTimeout' with a defined delay, or integrate a state management solution that handles such updates more predictably. <br> **Code Suggestion**: 
 ```
 - queueMicrotask(() => {
 + setTimeout(() => {
 ```